File: backend/InfoTrack.Domain/Locations/Location.cs
namespace InfoTrack.Domain.Locations;

public sealed record Location
{
    private const int MaxNameLength = 80;

    public Location(string name)
    {
        if (string.IsNullOrWhiteSpace(name))
        {
            throw new ArgumentException("Location is required.", nameof(name));
        }

        var normalizedName = name.Trim();
        if (normalizedName.Length > MaxNameLength)
        {
            throw new ArgumentException($"Location cannot exceed {MaxNameLength} characters.", nameof(name));
        }

        if (normalizedName.Any(character => !IsSupportedLocationCharacter(character)))
        {
            throw new ArgumentException("Location contains unsupported characters.", nameof(name));
        }

        Name = normalizedName;
    }

    public string Name { get; }

    private static bool IsSupportedLocationCharacter(char character)
    {
        return char.IsLetterOrDigit(character) ||
               character is ' ' or '-' or '\'' or '.';
    }
}

File: frontend/src/utils/location.ts
export const MAX_LOCATION_LENGTH = 80;

export function normalizeLocationInput(value: string): string {
  return value.trim();
}

export function validateLocationInput(value: string): string | null {
  const normalized = normalizeLocationInput(value);
  if (!normalized) {
    return null;
  }

  if (normalized.length > MAX_LOCATION_LENGTH) {
    return `Location cannot exceed ${MAX_LOCATION_LENGTH} characters.`;
  }

  return null;
}
